### PR TITLE
Add support for Vim and Emacs key bindings

### PR DESCRIPTION
**Pull request by teppah**
This PR adds support for Vim and Emacs key bindings to the editor.

A dropdown has been added in the control bar to allow for selecting the editor mode. 
control bar

The default option is "Default", which means no special key bindings are applied. It sets `keyboardHandler` on `AceEditor` to `"windows"`. 
dropdown
The dropdown contains two other modes: "Vim" and "Emacs". They respectively set `keyboardHandler` on the `AceEditor` component to `"vim"` and `"emacs"`.

The currently selected mode is persisted in local storage and is loaded and set by `KeyboardHandlerDropdown`.

### CODE REVIEW

File: src/Editor/index.jsx [hunk 1]
```diff
@@ -4,15 +4,16 @@ import AceEditor from "react-ace";
 import { UndoManager } from "brace";
 import "brace/theme/idle_fingers";
 import "brace/ext/searchbox";
+import "ace-builds/src-noconflict/keybinding-vim";
 
 import "./styles.css";
 import "./ace-mode-george";
 import "./ace-auto-complete-george";
-
```
Comment (ShazzAmin) [inline src/Editor/index.jsx:11]: nit: don't remove this line

File: src/KeyboardHandlerDropdown/index.jsx [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+  handleChange = (e) => {
+    this.props.onKeyboardHandlerChange(e.target.value);
+    LocalStorage.set("keyboard_handler", e.target.value);
+  };
+
+  render() {
+    return (
+      <div className="keyboard-handler-dropdown">
+        <label htmlFor="keyboard-handler">Key bindings:</label>
```
Comment (ShazzAmin) [inline src/KeyboardHandlerDropdown/index.jsx:41]: remove the label and just name the options "Default Keybindings", "Vim Keybindings" & "Emacs Keybindings". should make the UI feel less cluttered imo

File: src/KeyboardHandlerDropdown/styles.scss [hunk 3]
```diff
@@ -0,0 +1,13 @@
+@import "../common/variables";
+
+.keyboard-handler-dropdown {
```
Comment (ShazzAmin) [inline src/KeyboardHandlerDropdown/styles.scss:3]: can remove this when you remove label since there won't be a need for the wrapper div

**Review by ShazzAmin — CHANGES_REQUESTED** [review]
Thanks for taking this on! Just a few small tweaks and some nits.

File: src/Editor/index.jsx [hunk 4]
```diff
@@ -4,6 +4,8 @@ import AceEditor from "react-ace";
 import { UndoManager } from "brace";
 import "brace/theme/idle_fingers";
 import "brace/ext/searchbox";
+import "ace-builds/src-noconflict/keybinding-vim";
+import "ace-builds/src-noconflict/keybinding-emacs";
```
Comment (teppah) [inline src/Editor/index.jsx:8]: added the import for emacs keybindings from `ace-builds`, so now it should actually work

File: src/KeyboardHandlerDropdown/styles.scss (hunk 3 above)
Comment (teppah) [reply src/KeyboardHandlerDropdown/styles.scss:3]: renamed `.keyborad-handler-dropdown__select` -> `.keyborad-handler-dropdown`

**Comment by teppah** [comment]
Label has been removed and Emacs bindings should work properly now
image

**Review by ShazzAmin — APPROVED** [review]
Nice work! This is great.

**Pull request merged**